pandas/core/internals/blocks.py

```diff
@@ -739,49 +751,6 @@ def replace_list(
             rb = new_rb
         return rb
 
-    @final
-    def _replace_coerce(
-        self,
-        to_replace,
-        value,
-        mask: npt.NDArray[np.bool_],
-        inplace: bool = True,
-        regex: bool = False,
-    ) -> list[Block]:
-        """
-        Replace value corresponding to the given boolean array with another
-        value.
-
-        Parameters
-        ----------
-        to_replace : object or pattern
-            Scalar to replace or regular expression to match.
-        value : object
-            Replacement object.
-        mask : np.ndarray[bool]
-            True indicate corresponding element is ignored.
-        inplace : bool, default True
-            Perform inplace modification.
-        regex : bool, default False
-            If true, perform regular expression substitution.
-
-        Returns
-        -------
-        List[Block]
-        """
-        if should_use_regex(regex, to_replace):
-            return self._replace_regex(
-                to_replace,
-                value,
-                inplace=inplace,
-                convert=False,
```

Comment:
this is the convert for the regex case that has been ignored since https://github.com/pandas-dev/pandas/pull/40082